pkg/controller/controller_test.go

```diff
@@ -1090,8 +1198,8 @@ func provisionTestcases() (int64, map[string]provisioningTestcase) {
 	}
 	vacName := "test-vac"
 	return requestedBytes, map[string]provisioningTestcase{
-		"normal provision": {
-			volOpts: controller.ProvisionOptions{
+		"normal provision": *provisionTestcaseBuilder().volOpts(
```

Comment:
I think it became a little bit harder to read with chaining, compare:

```
makeProvisionTestcaseBuilder().volOpts(
  ...
).foo(
  ...
).bar(
  ...
).build()
```

vs

```
makeProvisionTestcaseBuilder(
  withFoo(),
  withBar(),
  withVolOpts()
).build()
```

Reply:
OK. Let me implement it using the functional option pattern.

Reply:
I wrote the first three test cases with the functional option pattern.

As you said, the test case setup part in `buildProvisionTestcase()` has become a bit simpler. On the other hand, the functions for the functional option pattern seem to be a bit more complicated than the methods for method chaining. That is why I chose the method chaining. However, I guess the simplicity of the test case setup part might be more important. So, I will use the functional option pattern for the rest of the test cases.

What do you think?